test/topology_custom/test_mv_backlog.py

```diff
@@ -0,0 +1,112 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+from test.pylib.manager_client import ManagerClient
+
+import asyncio
+import pytest
+import time
+import logging
+from test.topology.conftest import skip_mode
+from cassandra.cqltypes import Int32Type
+import requests
+import re
+
+logger = logging.getLogger(__name__)
+
+async def wait_for_view(cql, name, node_count):
+    deadline = time.time() + 120
+    while time.time() < deadline:
+        done = await cql.run_async(f"SELECT COUNT(*) FROM system_distributed.view_build_status WHERE status = 'SUCCESS' AND view_name = '{name}' ALLOW FILTERING")
+        if done[0][0] == node_count:
+            return
+        else:
+            time.sleep(0.2)
+    raise Exception("Timeout waiting for views to build")
+
+
+def get_metrics(addr):
+    # The Prometheus API is on port 9180, and always http
+    addr = 'http://' + addr + ':9180/metrics'
+    resp = requests.get(addr)
+    if resp.status_code != 200:
+        pytest.skip('Metrics port 9180 is not available')
+    response = requests.get(addr)
+    assert response.status_code == 200
+    return response.text
+
+def get_metric(metrics, name, requested_labels=None):
+    the_metrics = get_metrics(metrics)
+    total = 0.0
+    lines = re.compile('^'+name+'{.*$', re.MULTILINE)
+    for match in re.findall(lines, the_metrics):
+        a = match.split()
+        metric = a[0]
+        val = float(a[1])
+        # Check if match also matches the requested labels
+        if requested_labels:
+            # we know metric begins with name{ and ends with } - the labels
+            # are what we have between those
+            got_labels = metric[len(name)+1:-1].split(',')
+            # Check that every one of the requested labels is in got_labels:
+            for k, v in requested_labels.items():
+                if not f'{k}="{v}"' in got_labels:
+                    # No match for requested label, skip this metric (python
+                    # doesn't have "continue 2" so let's just set val to 0...
+                    val = 0
+                    break
+        total += float(val)
+    return total
+
+def get_replicas(cql, key):
+    return cql.cluster.metadata.get_replicas("ks", Int32Type.serialize(key, cql.cluster.protocol_version))
```

Comment:
I think this code will break with tablets :-(
The replica depends on the table. In particular, the base and view replicas don't need to be the same for the same integer key.
If I understand correctly, you are looking for  awrite where the **view** replica is on the second node, right?

Reply:
Right, this will most likely not work with tablets. I saw your approach in `topology_experimental_raft/test_mv_tablets` that achieves the same thing in tablets (write where the **view** replica is on a different node than the base) and I could definitely use that, but on the other hand, that approach does not work with vnodes. Should we have tests for both (this one would need to be changed so that it only runs when not using tablets) or would just the test working on tablets be enough (this patch will be only in 6.1+ where I believe tablets are the default)?

Reply:
It would be great if this test worked with tablets. Not sure if and when, but I suppose we would like to deprecate vnodes at some point. We will probably need to do these kinds of tricks in future tests, so learning how to do it with tablets right now might be a good time investment.

Reply:
I'm not sure, @sylwiaszunejko ?
It would be good to change `metadata.get_replicas` to make it work both with vnodes and tablets, but there are issues with that:
- it breaks backwards compatibility, which we should not do in Python Driver.
- Tablets are collected by driver in a lazy way (driver may not be even aware that given table is tablet-table, or even if it is it may not have a tablet for given token), so the method would be best-effort only. 

Also, wdym by `the shard of a token`? I was under the impression that there is no such things with tablets, only with vnodes - because in tablets a token may be replicated on different shards of different nodes.

Reply:
> Also, wdym by `the shard of a token`? I was under the impression that there is no such things with tablets, only with vnodes - because in tablets a token may be replicated on different shards of different nodes.

Sorry, a mental shortcut - I meant, I'd like to get a list of replicas where "replica" means a pair: identifier of the node and the owning shard.

Reply:
Perhaps we could take a look into the system tablets that describe tablets ourselves... We could have a helper function in the tests for that. Maybe it already exists (didn't check)? No need to involve the driver, then.

Reply:
I am not sure if I understand the question, as @Lorak-mmk saidva token may be replicated on different shards of different nodes. The tablet information is stored in https://github.com/scylladb/python-driver/blob/master/cassandra/metadata.py#L130 and this information is only fetched lazily.

Reply:
The driver definitely is supposed to know which node+shard a token sits on, otherwise it wouldn't have been a token-aware driver. However, there's probably no guarantee that this knowledge is immediate.

I think it is fine for a Scylla-only test (unfortunately, all "topology" framework tests are Scylla-only) to have code which uses the Scylla-specific system tables to read the current node+shard replicas for a token when tablets are enabled. In fact, we have such code already (see for example get_tablet_replicas() in test_mv_tablets.py, and there's also a more restricted version in pylib/). We don't need to get this capability from the driver if it's not natural there.

Reply:
Using the methods from `test_mv_tablets.py` I changed the test to use tablets instead in the rebase